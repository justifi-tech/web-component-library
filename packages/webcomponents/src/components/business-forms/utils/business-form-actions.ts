import { ComponentErrorSeverity } from "../../../api/ComponentError";
import { getErrorCode, getErrorMessage } from "../../../api/services/utils";

export const makeGetBusiness = ({ authToken, businessId, service }) =>
  async ({ onSuccess, onError }) => {
    try {
      const response = await service.fetchBusiness(authToken, businessId);

      if (!response.error) {
        const business = response.data;
        onSuccess({ business });
      } else {
        const responseError = getErrorMessage(response.error);
        const code = getErrorCode(response.error?.code);
        return onError({
          error: responseError,
          code,
          severity: ComponentErrorSeverity.ERROR,
        });
      }
    } catch (error) {
      const code = getErrorCode(error?.code);
      return onError({
        error: error.message || error,
        code,
        severity: ComponentErrorSeverity.ERROR,
      });
    }
  };

export const makePatchBusiness = ({ authToken, businessId, service }) =>
  async ({ business, onSuccess, onError }) => {
    try {
      const response = await service.patchBusiness(authToken, businessId, business);

      if (!response.error) {
        onSuccess(response);
      } else {
        const responseError = getErrorMessage(response.error);
        const code = getErrorCode(response.error?.code);
        return onError({
          error: responseError,
          code,
          severity: ComponentErrorSeverity.ERROR,
        });
      }
    } catch (error) {
      const code = getErrorCode(error?.code);
      return onError({
        error: error.message || error,
        code,
        severity: ComponentErrorSeverity.ERROR,
      });
    }
  };
  