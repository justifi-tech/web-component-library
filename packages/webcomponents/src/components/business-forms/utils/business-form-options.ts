import { BusinessClassification } from "../../../api/Business";

interface BusinessFormOption {
  label: string;
  value: string;
}

export const businessClassificationOptions: BusinessFormOption[] = [
  {
    label: 'Choose business classification',
    value: '',
  },
  {
    label: 'Sole Proprietor',
    value: BusinessClassification.sole_proprietor,
  },
  {
    label: 'Partnership',
    value: BusinessClassification.partnership,
  },
  {
    label: 'Corporation',
    value: BusinessClassification.corporation,
  },
  {
    label: 'Public Company',
    value: BusinessClassification.public_company,
  },
  {
    label: 'Limited',
    value: BusinessClassification.limited,
  },
  {
    label: 'Non Profit',
    value: BusinessClassification.non_profit,
  },
  {
    label: 'Government',
    value: BusinessClassification.government,
  },
];

export const businessServiceReceivedOptions: BusinessFormOption[] = [
  {
    label: 'Within 7 days',
    value: 'Within 7 days',
  },
  {
    label: 'Within 30 days',
    value: 'Within 30 days',
  },
  {
    label: 'Within 90 days',
    value: 'Within 90 days',
  },
  {
    label: 'Within 120 days',
    value: 'Within 120 days',
  },
  {
    label: 'More than 120 days',
    value: 'More than 120 days',
  }
];

export const recurringPaymentsOptions: BusinessFormOption[] = [
  {
    label: 'Yes',
    value: 'Yes',
  },
  {
    label: 'No',
    value: 'No',
  }
];

export const seasonalBusinessOptions: BusinessFormOption[] = [
  {
    label: 'Yes. The majority of the business revenue is generated in 3-6 months.',
    value: 'Yes',
  },
  {
    label: 'No. The business revenue is generated evenly throughout the year.',
    value: 'No',
  }
];

export const bankAccountTypeOptions: BusinessFormOption[] = [
  {
    label: 'Checking',
    value: 'checking',
  },
  {
    label: 'Savings',
    value: 'savings',
  }
];