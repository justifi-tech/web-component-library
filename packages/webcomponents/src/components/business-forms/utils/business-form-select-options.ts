type BusinessType =
  | 'for_profit'
  | 'non_profit'
  | 'government_entity'
  | 'individual'
  | '';


export const BusinessTypeOptions: { label: string; value: BusinessType }[] = [
  {
    label: 'Choose business type',
    value: '',
  },
  {
    label: 'Individual',
    value: 'individual',
  },
  {
    label: 'For Profit',
    value: 'for_profit',
  },
  {
    label: 'Non Profit',
    value: 'non_profit',
  },
  {
    label: 'Government Entity',
    value: 'government_entity',
  },
];
