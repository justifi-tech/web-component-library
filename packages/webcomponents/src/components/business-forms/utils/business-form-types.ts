export interface BusinessFormSubmitEvent {
  data: any;
  metadata?: BusinessFormEventMetaData;
}

export interface BusinessFormClickEvent {
  data?: any;
  name: BusinessFormClickActions;
}

export interface BusinessFormServerErrorEvent {
  data?: any;
  message: BusinessFormServerErrors;
}

// For right now we can use MetaData to track the step that was just completed during a submit event. 
// Eventually we can find other uses for MetaData to present to the implementer. 
interface BusinessFormEventMetaData {
  completedStep?: BusinessFormStep;
}

type BusinessFormStep = 'coreInfo' | 'legalAddress' | 'additionalQuestions' | 'representative' | 'owners';

export enum BusinessFormServerErrors {
  fetchData = 'Error retrieving business data',
  patchData = 'Error updating business data'
}

export enum BusinessFormClickActions {
  nextStep = 'nextStep',
  previousStep = 'previousStep',
  submit = 'submit'
}

export interface OwnerFormSubmitEvent {
  data: any;
  metadata?: any;
}

export interface OwnerFormClickEvent {
  data?: any;
  name: OwnerFormClickActions;
}

export interface OwnerFormServerErrorEvent {
  data?: any;
  message: OwnerFormServerErrors;
}

export enum OwnerFormServerErrors {
  fetchData = 'Error retrieving owner data',
  patchData = 'Error updating owner data',
  postData = 'Error adding owner data'
}

export enum OwnerFormClickActions {
  removeOwner = 'removeOwner',
  addOwner = 'addOwner',
  addOwnerForm = 'addOwnerForm',
  updateOwner = 'updateOwner'
}
