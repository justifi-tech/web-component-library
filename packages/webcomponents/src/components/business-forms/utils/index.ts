export * from './business-form-options';
export * from './helpers';
export * from './input-handlers';
export * from './event-types';
export * from './country-form-helpers';
