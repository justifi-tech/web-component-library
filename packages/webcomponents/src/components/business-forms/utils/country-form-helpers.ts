import { getRegionOptions, getRegionLabel, getPostalCodeLabel, CountryCode } from '../../../utils/address-form-helpers';
import { numberOnlyHandler } from '../../../ui-components/form/utils';

export interface CountryFormConfig {
  regionOptions: Array<{ label: string; value: string }>;
  regionLabel: string;
  postalCodeLabel: string;
  postalCodeConfig: {
    maxLength: number;
    keyDownHandler?: (event: KeyboardEvent) => void;
  };
  postalCodeHelpText: string;
}

/**
 * Gets country-specific form configuration including region options, labels, and postal code settings
 * @param country - The country code (e.g., 'USA', 'CAN')
 * @returns Object containing regionOptions, regionLabel, postalCodeLabel, and postalCodeConfig
 */
export function getCountryFormConfig(country?: string): CountryFormConfig {
  const regionOptions = getRegionOptions(country);
  const regionLabel = getRegionLabel(country);
  const postalCodeLabel = getPostalCodeLabel(country);

  // Configure postal code input based on country
  const postalCodeConfig = country === CountryCode.CAN ? {
    maxLength: 7, // A1A 1A1 with space
    keyDownHandler: undefined, // Allow letters for Canadian postal codes
  } : {
    maxLength: 10, // 12345-6789 for US extended zip
    keyDownHandler: numberOnlyHandler,
  };

  // Configure help text based on country
  const postalCodeHelpText = country === CountryCode.CAN ? 'Format: A1A 1A1' : 'Format: 12345 or 12345-6789';

  return {
    regionOptions,
    regionLabel,
    postalCodeLabel,
    postalCodeConfig,
    postalCodeHelpText,
  };
} 