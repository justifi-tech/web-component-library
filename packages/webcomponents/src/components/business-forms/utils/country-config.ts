import { CountryCode } from '../../../utils/country-codes';
import StateOptions from '../../../utils/state-options';
import ProvinceOptions from '../../../utils/province-options';
import { businessClassificationOptions } from './business-form-options';

export const countryLabels = {
	USA: {
		stateLabel: 'State',
		postalLabel: 'Zip Code',
		identityLabel: 'SSN',
		identityHelpText: 'Enter your full Social Security Number. It is required for Federal OFAC check.',
		taxIdLabel: 'Tax ID (EIN or SSN)',
		taxIdHelpText: "Employer Identification Numbers (EINs) are nine digits. The federal tax identification number/EIN issued to you by the IRS. It can be found on your tax returns. Enter value without dashes.",
	},
	CAN: {
		stateLabel: 'Province',
		postalLabel: 'Postal Code',
		identityLabel: 'SIN',
		identityHelpText: 'Enter your full Social Insurance Number.',
		taxIdLabel: 'Business Number (BN)',
		taxIdHelpText: 'Business Numbers (BN) are nine digits. Enter value without spaces or dashes.',
	},
} as const;

export const countryOptions = {
	USA: {
		stateOptions: StateOptions,
	},
	CAN: {
		stateOptions: ProvinceOptions,
	},
} as const;

export const countryValidation = {
	USA: {
		postalRegex: /^[0-9]{5}(-[0-9]{4})?$/,
		identityDigits: 9,
		taxIdDigits: 9,
	},
	CAN: {
		postalRegex: /^[A-Za-z]\d[A-Za-z][ -]?\d[A-Za-z]\d$/,
		identityDigits: 9,
		taxIdDigits: 9,
	},
} as const;

export const getStateValues = (country: CountryCode): string[] =>
	(countryOptions[country].stateOptions || []).map((o: any) => o.value);

// Business structures (placeholder for CAN duplicates USA for now)
export const businessClassificationOptionsUSA = businessClassificationOptions;
export const businessClassificationOptionsCAN = businessClassificationOptions;