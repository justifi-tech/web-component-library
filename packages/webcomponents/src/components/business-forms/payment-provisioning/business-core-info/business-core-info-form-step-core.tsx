import { Component, Host, h, Prop, State, Method, Event, EventEmitter } from '@stencil/core';
import { businessCoreInfoSchema } from '../../schemas/business-core-info-schema';
import { FormController } from '../../../form/form';
import { CoreBusinessInfo, ICoreBusinessInfo } from '../../../../api/Business';
import { BusinessFormSubmitEvent } from '../../utils/business-form-types';
import { ComponentError } from '../../../../api/ComponentError';
import { businessClassificationOptions } from '../../utils/business-form-options';
import { PHONE_MASKS } from '../../../../utils/form-input-masks';
import { parseCoreInfo } from '../../utils/payload-parsers';
import { flattenNestedObject } from '../../../../utils/utils';
import { numberOnlyHandler } from '../../../form/utils';

@Component({
  tag: 'justifi-business-core-info-form-step-core',
})
export class BusinessCoreInfoFormStepCore {
  @State() formController: FormController;
  @State() errors: any = {};
  @State() coreInfo: ICoreBusinessInfo = {};

  @Prop() authToken: string;
  @Prop() businessId: string;
  @Prop() getBusiness: Function;
  @Prop() patchBusiness: Function;
  @Prop() allowOptionalFields?: boolean;

  @Event({ bubbles: true }) submitted: EventEmitter<BusinessFormSubmitEvent>;
  @Event({ bubbles: true }) formLoading: EventEmitter<boolean>;
  @Event({ eventName: 'error-event', bubbles: true }) errorEvent: EventEmitter<ComponentError>;

  @Method()
  async validateAndSubmit({ onSuccess }) {
    this.formController.validateAndSubmit(() => this.sendData(onSuccess));
  };

  get patchPayload() {
    let formValues = parseCoreInfo(flattenNestedObject(this.formController.values.getValue()));
    return JSON.stringify(formValues);
  }

  componentWillLoad() {
    this.getBusiness && this.getData();
    this.formController = new FormController(businessCoreInfoSchema(this.allowOptionalFields));
  }

  componentDidLoad() {
    this.formController.errors.subscribe(errors => {
      this.errors = { ...errors };
    });
  }

  private getData = () => {
    this.formLoading.emit(true);
    this.getBusiness({
      onSuccess: (response) => {
        this.coreInfo = new CoreBusinessInfo(response.data);
        this.formController.setInitialValues({ ...this.coreInfo });
      },
      onError: ({ error, code, severity }) => {
        this.errorEvent.emit({
          message: error,
          errorCode: code,
          severity: severity
        });
      },
      final: () => this.formLoading.emit(false)
    });
  }

  private sendData = (onSuccess: () => void) => {
    let submittedData;
    this.formLoading.emit(true);
    this.patchBusiness({
      payload: this.patchPayload,
      onSuccess: (response) => {
        submittedData = response;
        onSuccess();
      },
      onError: ({ error, code, severity }) => {
        submittedData = error;
        this.errorEvent.emit({
          message: error,
          errorCode: code,
          severity: severity
        });
      },
      final: () => {
        this.submitted.emit({ data: { submittedData } });
        this.formLoading.emit(false)
      }
    });
  }

  inputHandler = (name: string, value: string) => {
    this.formController.setValues({
      ...this.formController.values.getValue(),
      [name]: value,
    });
  }

  render() {
    const coreInfoDefaultValue = this.formController.getInitialValues();

    return (
      <Host exportparts='label,input,input-invalid'>
        <form>
          <fieldset>
            <legend>General Info</legend>
            <hr />
            <div class='row gy-3'>
              <div class='col-12'>
                <form-control-text
                  name='legal_name'
                  label='Legal Name'
                  defaultValue={coreInfoDefaultValue.legal_name}
                  errorText={this.errors.legal_name}
                  inputHandler={this.inputHandler}
                />
              </div>
              <div class='col-12'>
                <form-control-text
                  name='doing_business_as'
                  label='Doing Business As (DBA)'
                  defaultValue={coreInfoDefaultValue.doing_business_as}
                  errorText={this.errors.doing_business_as}
                  inputHandler={this.inputHandler}
                />
              </div>
              <div class='col-12 col-md-8'>
                <form-control-select
                  name='classification'
                  label='Business Classification'
                  options={businessClassificationOptions}
                  defaultValue={coreInfoDefaultValue.classification}
                  errorText={this.errors.classification}
                  inputHandler={this.inputHandler}
                />
              </div>
              <div class='col-12 col-md-4'>
                <form-control-date
                  name='date_of_incorporation'
                  label='Date of Incorporation'
                  defaultValue={coreInfoDefaultValue.date_of_incorporation}
                  errorText={this.errors.date_of_incorporation}
                  inputHandler={this.inputHandler}
                />
              </div>
              <div class='col-12 col-md-6'>
                <form-control-text
                  name='industry'
                  label='Industry'
                  defaultValue={coreInfoDefaultValue.industry}
                  errorText={this.errors.industry}
                  inputHandler={this.inputHandler}
                />
              </div>
              <div class='col-12 col-md-6'>
                <form-control-text
                  name='tax_id'
                  label='Tax ID (EIN or SSN)'
                  defaultValue={coreInfoDefaultValue.tax_id}
                  errorText={this.errors.tax_id}
                  inputHandler={this.inputHandler}
                  keyDownHandler={numberOnlyHandler}
                  maxLength={9}
                />
              </div>
              <div class='col-12'>
                <form-control-text
                  name='website_url'
                  label='Website URL'
                  defaultValue={coreInfoDefaultValue.website_url}
                  errorText={this.errors.website_url}
                  inputHandler={this.inputHandler}
                />
              </div>
              <div class='col-12 col-md-6'>
                <form-control-text
                  name='email'
                  label='Email Address'
                  defaultValue={coreInfoDefaultValue.email}
                  errorText={this.errors.email}
                  inputHandler={this.inputHandler}
                />
              </div>
              <div class='col-12 col-md-6'>
                <form-control-number-masked
                  name='phone'
                  label='Phone Number'
                  defaultValue={coreInfoDefaultValue.phone}
                  errorText={this.errors.phone}
                  inputHandler={this.inputHandler}
                  mask={PHONE_MASKS.US}
                />
              </div>
            </div>
          </fieldset>
        </form>
      </Host>
    );
  }
}
