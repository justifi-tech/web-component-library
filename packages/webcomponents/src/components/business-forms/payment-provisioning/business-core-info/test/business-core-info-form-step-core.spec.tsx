import { h } from '@stencil/core';
import { newSpecPage } from '@stencil/core/testing';
import { BusinessCoreInfoFormStepCore } from '../business-core-info-form-step-core';
import { CountryCode } from '../../../../../utils/country-codes';

describe('justifi-business-core-info-form-step-core', () => {
  const mockGetBusiness = jest.fn();
  const mockPatchBusiness = jest.fn();

  beforeEach(() => {
    mockGetBusiness.mockClear();
    mockPatchBusiness.mockClear();
  });

  describe('Country-specific tax ID label', () => {
    test('renders USA tax ID label and help text', async () => {
      const page = await newSpecPage({
        components: [BusinessCoreInfoFormStepCore],
        template: () => (
          <justifi-business-core-info-form-step-core
            businessId="biz_123"
            getBusiness={mockGetBusiness}
            patchBusiness={mockPatchBusiness}
            country={CountryCode.USA}
          />
        ),
      });
      
      expect(page.root).toMatchSnapshot();
    });

    test('renders CAN Business Number (BN) label and help text', async () => {
      const page = await newSpecPage({
        components: [BusinessCoreInfoFormStepCore],
        template: () => (
          <justifi-business-core-info-form-step-core
            businessId="biz_123"
            getBusiness={mockGetBusiness}
            patchBusiness={mockPatchBusiness}
            country={CountryCode.CAN}
          />
        ),
      });
      
      expect(page.root).toMatchSnapshot();
    });
  });

  describe('Tax ID toggle behavior', () => {
    test('shows regular input when component loads with no data', async () => {
      mockGetBusiness.mockImplementation(({ final }) => final?.());

      const page = await newSpecPage({
        components: [BusinessCoreInfoFormStepCore],
        template: () => (
          <justifi-business-core-info-form-step-core
            businessId="biz_123"
            getBusiness={mockGetBusiness}
            patchBusiness={mockPatchBusiness}
            country={CountryCode.USA}
          />
        ),
      });
      
      const taxIdField = page.root.querySelector('toggleable-field[fieldname="tax_id"]');
      expect(taxIdField).toBeTruthy();
      // When no data is loaded, readOnlyValue should be null/undefined
      expect(taxIdField.getAttribute('readonlyvalue')).toBe(null);
      expect(taxIdField.getAttribute('maxlength')).toBe('9');
    });

    test('shows read-only display when tax_id_last4 is present', async () => {
      // Mock getBusiness to return data with tax_id_last4
      mockGetBusiness.mockImplementation(({ onSuccess, final }) => {
        onSuccess({
          data: {
            tax_id: '123456789',
            tax_id_last4: '6789'
          }
        });
        final?.();
      });
      
      const page = await newSpecPage({
        components: [BusinessCoreInfoFormStepCore],
        template: () => (
          <justifi-business-core-info-form-step-core
            businessId="biz_123"
            getBusiness={mockGetBusiness}
            patchBusiness={mockPatchBusiness}
            country={CountryCode.USA}
          />
        ),
      });
      
      // Wait for component to load and call getBusiness
      await page.waitForChanges();
      const taxIdField = page.root.querySelector('toggleable-field[fieldname="tax_id"]');
      expect(taxIdField).toBeTruthy();
      expect(taxIdField.getAttribute('readonlyvalue')).toBe('6789');
    });
  });

  describe('Form component behavior', () => {
    test('uses toggleable-field instead of form-control-text', async () => {
      mockGetBusiness.mockImplementation(({ final }) => final?.());

      const page = await newSpecPage({
        components: [BusinessCoreInfoFormStepCore],
        template: () => (
          <justifi-business-core-info-form-step-core
            businessId="biz_123"
            getBusiness={mockGetBusiness}
            patchBusiness={mockPatchBusiness}
            country={CountryCode.USA}
          />
        ),
      });
      
      const taxIdField = page.root.querySelector('toggleable-field[fieldname="tax_id"]');
      expect(taxIdField).toBeTruthy();
      
      // Should not find the old form-control-text
      const textTaxIdField = page.root.querySelector('form-control-text[name="tax_id"]');
      expect(textTaxIdField).toBeNull();
    });
  });

  describe('Data loading behavior', () => {
    test('calls getBusiness on component load when provided', async () => {
      await newSpecPage({
        components: [BusinessCoreInfoFormStepCore],
        template: () => (
          <justifi-business-core-info-form-step-core
            businessId="biz_123"
            getBusiness={mockGetBusiness}
            patchBusiness={mockPatchBusiness}
            country={CountryCode.USA}
          />
        ),
      });
      
      expect(mockGetBusiness).toHaveBeenCalledTimes(1);
    });
  });
});