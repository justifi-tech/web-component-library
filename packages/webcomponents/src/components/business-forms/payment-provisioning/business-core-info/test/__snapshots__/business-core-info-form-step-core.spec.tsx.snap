// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`justifi-business-core-info-form-step-core Country-specific tax ID label renders CAN Business Number (BN) label and help text 1`] = `
<justifi-business-core-info-form-step-core>
  <div class="gap-5 row">
    <div class="col-12">
      <div aria-hidden="true" class="skeleton undefined" part="skeleton" style="width: 40%; height: 40px;"></div>
    </div>
    <div class="gap-3 row">
      <div class="col-12">
        <div aria-hidden="true" class="skeleton undefined" part="skeleton" style="width: 25%;"></div>
      </div>
      <div class="col-12">
        <div aria-hidden="true" class="skeleton undefined" part="skeleton" style="width: 100%;"></div>
      </div>
      <div class="col-12 d-flex gap-3">
        <div aria-hidden="true" class="skeleton undefined" part="skeleton" style="width: 50%;"></div>
        <div aria-hidden="true" class="skeleton undefined" part="skeleton" style="width: 50%;"></div>
      </div>
      <div class="col-12 d-flex gap-3">
        <div aria-hidden="true" class="skeleton undefined" part="skeleton" style="width: 65%;"></div>
        <div aria-hidden="true" class="skeleton undefined" part="skeleton" style="width: 35%;"></div>
      </div>
    </div>
    <div class="gap-3 row">
      <div class="col-12">
        <div aria-hidden="true" class="skeleton undefined" part="skeleton" style="width: 25%;"></div>
      </div>
      <div class="col-12">
        <div aria-hidden="true" class="skeleton undefined" part="skeleton" style="width: 100%;"></div>
      </div>
      <div class="col-12 d-flex gap-3">
        <div aria-hidden="true" class="skeleton undefined" part="skeleton" style="width: 50%;"></div>
        <div aria-hidden="true" class="skeleton undefined" part="skeleton" style="width: 50%;"></div>
      </div>
      <div class="col-12 d-flex gap-3">
        <div aria-hidden="true" class="skeleton undefined" part="skeleton" style="width: 65%;"></div>
        <div aria-hidden="true" class="skeleton undefined" part="skeleton" style="width: 35%;"></div>
      </div>
    </div>
    <div class="gap-3 row">
      <div class="col-12">
        <div aria-hidden="true" class="skeleton undefined" part="skeleton" style="width: 25%;"></div>
      </div>
      <div class="col-12">
        <div aria-hidden="true" class="skeleton undefined" part="skeleton" style="width: 100%;"></div>
      </div>
      <div class="col-12 d-flex gap-3">
        <div aria-hidden="true" class="skeleton undefined" part="skeleton" style="width: 50%;"></div>
        <div aria-hidden="true" class="skeleton undefined" part="skeleton" style="width: 50%;"></div>
      </div>
      <div class="col-12 d-flex gap-3">
        <div aria-hidden="true" class="skeleton undefined" part="skeleton" style="width: 65%;"></div>
        <div aria-hidden="true" class="skeleton undefined" part="skeleton" style="width: 35%;"></div>
      </div>
    </div>
  </div>
</justifi-business-core-info-form-step-core>
`;

exports[`justifi-business-core-info-form-step-core Country-specific tax ID label renders USA tax ID label and help text 1`] = `
<justifi-business-core-info-form-step-core>
  <div class="gap-5 row">
    <div class="col-12">
      <div aria-hidden="true" class="skeleton undefined" part="skeleton" style="width: 40%; height: 40px;"></div>
    </div>
    <div class="gap-3 row">
      <div class="col-12">
        <div aria-hidden="true" class="skeleton undefined" part="skeleton" style="width: 25%;"></div>
      </div>
      <div class="col-12">
        <div aria-hidden="true" class="skeleton undefined" part="skeleton" style="width: 100%;"></div>
      </div>
      <div class="col-12 d-flex gap-3">
        <div aria-hidden="true" class="skeleton undefined" part="skeleton" style="width: 50%;"></div>
        <div aria-hidden="true" class="skeleton undefined" part="skeleton" style="width: 50%;"></div>
      </div>
      <div class="col-12 d-flex gap-3">
        <div aria-hidden="true" class="skeleton undefined" part="skeleton" style="width: 65%;"></div>
        <div aria-hidden="true" class="skeleton undefined" part="skeleton" style="width: 35%;"></div>
      </div>
    </div>
    <div class="gap-3 row">
      <div class="col-12">
        <div aria-hidden="true" class="skeleton undefined" part="skeleton" style="width: 25%;"></div>
      </div>
      <div class="col-12">
        <div aria-hidden="true" class="skeleton undefined" part="skeleton" style="width: 100%;"></div>
      </div>
      <div class="col-12 d-flex gap-3">
        <div aria-hidden="true" class="skeleton undefined" part="skeleton" style="width: 50%;"></div>
        <div aria-hidden="true" class="skeleton undefined" part="skeleton" style="width: 50%;"></div>
      </div>
      <div class="col-12 d-flex gap-3">
        <div aria-hidden="true" class="skeleton undefined" part="skeleton" style="width: 65%;"></div>
        <div aria-hidden="true" class="skeleton undefined" part="skeleton" style="width: 35%;"></div>
      </div>
    </div>
    <div class="gap-3 row">
      <div class="col-12">
        <div aria-hidden="true" class="skeleton undefined" part="skeleton" style="width: 25%;"></div>
      </div>
      <div class="col-12">
        <div aria-hidden="true" class="skeleton undefined" part="skeleton" style="width: 100%;"></div>
      </div>
      <div class="col-12 d-flex gap-3">
        <div aria-hidden="true" class="skeleton undefined" part="skeleton" style="width: 50%;"></div>
        <div aria-hidden="true" class="skeleton undefined" part="skeleton" style="width: 50%;"></div>
      </div>
      <div class="col-12 d-flex gap-3">
        <div aria-hidden="true" class="skeleton undefined" part="skeleton" style="width: 65%;"></div>
        <div aria-hidden="true" class="skeleton undefined" part="skeleton" style="width: 35%;"></div>
      </div>
    </div>
  </div>
</justifi-business-core-info-form-step-core>
`;

exports[`justifi-business-core-info-form-step-core Form component behavior uses toggleable-field instead of form-control-text 1`] = `
<justifi-business-core-info-form-step-core>
  <form>
    <fieldset>
      <div class="align-items-center d-flex gap-2">
        <legend class="mb-0" part="heading-2 heading text color font-family">
          Business Information
        </legend>
      </div>
      <hr class="mt-2">
      <div class="gy-3 row">
        <div class="col-12">
          <form-control-text helptext="Enter this exactly as it appears on your tax records (don't use acronyms or abbreviations unless you registered that way)." label="Business Name" name="legal_name"></form-control-text>
        </div>
        <div class="col-12">
          <form-control-text helptext="Enter this exactly as it appears on your tax records (leave blank if you don't have a registered DBA/trade name)" label="Doing Business As (DBA)" name="doing_business_as"></form-control-text>
        </div>
        <div class="col-12 col-md-8">
          <form-control-select label="Business Classification" name="classification"></form-control-select>
        </div>
        <div class="col-12 col-md-4">
          <form-control-date label="Date of Incorporation" name="date_of_incorporation"></form-control-date>
        </div>
        <div class="col-12 col-md-6">
          <form-control-text helptext="Describe what you sell" label="Industry" name="industry"></form-control-text>
        </div>
        <div class="col-12 col-md-6">
          <form-control-number-masked helptext="Employer Identification Numbers (EINs) are nine digits. The federal tax identification number/EIN issued to you by the IRS. It can be found on your tax returns. Enter value without dashes." label="Tax ID (EIN or SSN)" mask="000000000" name="tax_id"></form-control-number-masked>
        </div>
        <div class="col-12">
          <form-control-text helptext="Don't have a live website? You can use your social media business page, app store link, or staging site URL." label="Business Website URL" name="website_url"></form-control-text>
        </div>
        <div class="col-12 col-md-6">
          <form-control-text label="Business Email Address" name="email"></form-control-text>
        </div>
        <div class="col-12 col-md-6">
          <form-control-number-masked label="Business Phone Number" mask="(000) 000-0000" name="phone"></form-control-number-masked>
        </div>
      </div>
    </fieldset>
  </form>
</justifi-business-core-info-form-step-core>
`;
